### PR TITLE
doc: peripherals: unify title strings

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Unify the peripheral documentation title strings to the format "<class> [(acronym)] [Bus]".

Including both the full name of the peripheral class and an acronym makes the documentation more user friendly as some of the acronyms are less well-known than others.

This is a follow-up to #56275.

### CODE REVIEW

**Review by martinjaeger — DISMISSED** [review]
Typo in the commit message (peri**hp**erals). Otherwise LGTM. Thanks!

**Review by teburd — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
> Typo in the commit message (peri**hp**erals). Otherwise LGTM. Thanks!

Fixed, thank you.

**Review by martinjaeger — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

File: doc/hardware/peripherals/adc.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _adc_api:
 
-ADC
-###
+Analog-to-Digial Converter (ADC)
```
Comment (anangl) [inline doc/hardware/peripherals/adc.rst:3]:
```suggestion
Analog-to-Digital Converter (ADC)
```

File: doc/hardware/peripherals/espi.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .. _espi_api:
 
-eSPI
-####
+Enhanced Serial Peripheral Interface (eSPI) bus
```
Comment (anangl) [inline doc/hardware/peripherals/espi.rst:3]:
```suggestion
Enhanced Serial Peripheral Interface (eSPI) Bus
```

File: doc/hardware/peripherals/adc.rst (hunk 1 above)
Comment (henrikbrixandersen) [reply doc/hardware/peripherals/adc.rst]: Ouch, thanks. Fixed.

File: doc/hardware/peripherals/espi.rst (hunk 2 above)
Comment (henrikbrixandersen) [reply doc/hardware/peripherals/espi.rst]: Fixed.

**Review by anangl — DISMISSED** [review]

**Review by martinjaeger — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]
This is very cool, thank you for proposing this change!

I would suggest the following additions:

- Gaussian & Neural Accelerator (GNA)
- Real-Time Clock (RTC)
- USB Type-C Port Controller (TCPC)
- MIPI Display Serial Interface (DSI)

Also, could you make sure index.rst remains (or rather, is) alphabetically ordered (e.g 1-Wire should now be showing up first).

Thank you!

**Review by fabiobaltieri — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Thank you!

**Pull request merged**